### DISCUSSION TITLE
Can Faces be manually added?

### DISCUSSION THREAD

**Discussion by ziggysid** [photoprism/photoprism#4644] [Q&A]
I have a photo of my two nephews sat side by side. Photoprism only seems to recognise that there is one face in the Photo. It lets me add a face for Nephew A, but I can't find anyway to add a Face for Nephew B.

**Comment by omerdduran** [comment]
Hi. PhotoPrism's core UI doesn't yet support manually adding face markers, though this feature is in the [roadmap](https://github.com/photoprism/photoprism/issues/1548). Meanwhile, you can:

1. Use the community-developed "PhotoPrismFaceMarker" Chrome extension: https://github.com/Blendan1/PhotoPrismFaceMarker 
**The browser extension mentioned is not affiliated with the official PhotoPrism project.**
2. Try adjusting detection sensitivity in your config:
- Lower PHOTOPRISM_FACE_SCORE
- Reduce PHOTOPRISM_FACE_SIZE
- Perform a complete rescan after changes

https://docs.photoprism.app/user-guide/organize/people/#when-a-face-was-not-detected
https://docs.photoprism.app/getting-started/config-options/#face-recognition

**Reply by ziggysid** [reply]
How to adjust the detection sensitivity; I read in the User Guide about Config Options, but I'm not sure where this is? It mentions three Config files: options.yml, defaults.yml and settings.yml. Which one do I need to amend and where can I find it?

Further reading suggests I might find these files in the /etc/photoprism folder, however, I can't find a "photoprism" folder in the /etc folder. Perhaps that means I have not installed photoprism to the default location, but if that's true, it was a misunderstanding on part rather than a conscious decision; meaning I don't know where else to look for these config files.

**Reply by omerdduran** [reply]
Hi. It's possible to adjust the face recognition parameters, but this is only recommended for experienced users. Each time you change these settings, you must first run the photoprism faces reset -f command, which removes all existing faces and people tags. This will prevent inconsistencies, but you'll lose all your existing face recognition data and have to start again. If you still want to continue setting it up, you can adjust the parameters from the compose file if you installed it with docker compose. If you did not install it with compose, it is recommended that you set it up with options.yml. Unless you have changed the configuration, the default mount/storage path for the options.yml file is /photoprism/storage/config/options.yml.

**Reply by ziggysid** [reply]
Not an experienced user. However, as I am trialling this Photoprisim, I've only loaded a few hundred photos and I don't mind if it needs to be re-indexed or I loose any of the faces previously assigned. I'd rather get the setup right, and know that it can meet my expectations, before committing thousands of my photos.

The installation is with Docker Compose.

Having tried different PHOTOPRISM_FACE_SIZE values from 20 to 100, and noticed no difference in the number of faces that are recognised, I'm of the conclusion I must be doing something wrong or missed a step perhaps?

1. stopped the Photoprism service using the `docker-compose stop` command
2. edited the docker-compose.yml file to include the line `PHOTOPRISM_FACE_SIZE: "20"`
3. restarted the Photoprism service using the `docker-compose up -d` command
4. Entered Photoprisim from the web browser and re-indexed the /originals directory.

As mentioned, I've repeated this with FACE_SIZE values from 20 to 100 and seen no difference in the number of faces detected. What did I miss?

**Reply by comprido** [reply]
> Hi. PhotoPrism's core UI doesn't yet support manually adding face markers, though this feature is in the [roadmap](https://github.com/photoprism/photoprism/issues/1548). Meanwhile, you can:
> 
> 1. Use the community-developed "PhotoPrismFaceMarker" Chrome extension: https://github.com/Blendan1/PhotoPrismFaceMarker
>    **The browser extension mentioned is not affiliated with the official PhotoPrism project.**

This seems to have stopped working with the latest version of PhotoPrism :( It was working perfectly before.


> 2. Try adjusting detection sensitivity in your config:
> 
> * Lower PHOTOPRISM_FACE_SCORE
> * Reduce PHOTOPRISM_FACE_SIZE
> * Perform a complete rescan after changes
> 
> https://docs.photoprism.app/user-guide/organize/people/#when-a-face-was-not-detected https://docs.photoprism.app/getting-started/config-options/#face-recognition

**Comment by omerdduran** [comment]
Hello! The steps you've taken seem correct. However, to verify if your face detection settings are actually applied, you can run this command inside the photoprism container to check the current configuration:

```bash
photoprism config
```

This will show all active settings, including FACE_SIZE, so you can confirm if your changes are being picked up by PhotoPrism. But only changing the “PHOTOPRISM_FACE_SIZE” setting may not change anything

**Reply by ziggysid** [reply]
After some more testing, reducing the `PHOTOPRISM_FACE_SCORE` was the most effective at capturing the extra faces. Thanks for your help with that.

You got me with the `photoprism config` command. It kept throwing up an error. In the end I found `docker config` is the command line that enabled me to check the face detection settings have been applied.

What does the `PHOTOPRISM_FACE_OVERLAP` do?

**Reply by omerdduran** [reply]
`PHOTOPRISM_FACE_OVERLAP` is a face detection parameter that controls how overlapping faces are handled during face detection. Specifically:
- It sets the face area overlap threshold as a percentage value (1-100)
- The default value appears to be 42 (meaning 42% overlap threshold)
- This parameter helps determine when multiple detected face regions should be considered as representing the same face versus separate faces
